File: 240604/양수 직사각형의 최대 크기/max-area-of-positive-rectangle.py
N, M = map(int, input().split())
board = [list(map(int, input().split())) for _ in range(N)]

def go(y,x,nn,mm):
    for r in range(y, y+nn):
        for c in range(x, x+mm):
            if board[r][c] <= 0:
                return False
    return True
ans = -1
for nn in range(N, 0, -1):
    for mm in range(M, 0, -1):
        for r in range(0, N-nn+1):
            for c in range(0, M -mm+1):
                if go(r,c,nn,mm):
                    ans = max(ans, nn * mm)

print(ans)

File: 240714/양수 직사각형의 최대 크기/max-area-of-positive-rectangle.java
import java.util.*;
import java.io.*; 

public class Main {
    public static int N, M;
    public static int[][] board;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        N = sc.nextInt();
        M = sc.nextInt();

        board = new int[N][M];

        for (int i = 0; i < N; i++){
            for (int j = 0; j < M; j++){
                board[i][j] = sc.nextInt();
            }
        }

        int ans = go();
        System.out.println(ans);

    }

    public static boolean out(int y, int x){
        return (y < 0 || N <= y || x < 0 || M <= x);
    }

    public static int go(){
        int result = -1;
        for (int y =0; y < N; y++){
            for (int x = 0; x < M; x++){
                for (int sy = 1; sy <= N; sy++){
                    for (int sx = 1; sx <= M; sx++){
                        // System.out.println(cal(y,x,sy,sx));
                       result = Math.max(result, cal(y,x,sy,sx));
                    }
                }
            }
        }
        return result;
    }

    public static int cal(int y, int x, int k, int l){
        int total = 0;
        for (int dy = 0; dy < k; dy++){
            for (int dx = 0; dx < l; dx++){
                if (out(y + dy, x + dx)){
                    return -1;
                }
                if (board[y+dy][x+dx] <= 0){
                    return -1;
                }
                total += 1;
            }
        }
        return total;
    }

}